### PR TITLE
Ft enquire buses info o6m b umpm

### PR DESCRIPTION
**Pull request by Egide138**
### What does this PR do?
• allow view  of buses info 
### Description of Task to be completed?

• user should be able to view buses's info to their respective routes
### How should this be manually tested?
• clone it to the local repo
• run npm install
• test the end points in postman: /api/bus/busInfo to get all buses info
.end point for getting one bus info:/api/bus/businfo/:id
.End point for updating info : /api/bus/businfo/:id
### What are the relevant trello id?
o6mBUmpm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,4 +77,4 @@
     "url": "https://github.com/atlp-rwanda/incredible-phantom-backend/issues"
   },
   "homepage": "https://github.com/atlp-rwanda/incredible-phantom-backend#readme"
-}
+}
```
Comment (raymond42) [inline package.json]: can you add an empty line at the end the file please?

File: src/controllers/busControllers.js [hunk 2]
```diff
@@ -131,3 +135,27 @@ export const deleteBus = async (req, res) => {
...
+      const response= await Bus.findOne({ where: { id },attributes: ['id','location',
+      'status','commuters','type','seats'] }
+          
+        );
+      successRes(res,200,res.__('All information you need about the bus'),response)
+    }catch(error){
+      console.log(error)
+    errorRes(res,500,res.__('error occured!Try again'))
+    }
+    }
```
Comment (raymond42) [inline src/controllers/busControllers.js]: same as here

File: src/controllers/usersController.js [hunk 3]
```diff
@@ -333,3 +333,16 @@ export const deleteNotification = async (req, res) => {
...
+
+
+
+
+
+  
+
+
+
+
```
Comment (raymond42) [inline src/controllers/usersController.js]: remove these empty lines. let's have only one empty line. you should follow eslint rules

File: documentation/swaggerDoc.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    apis: ["../src/routers/*.js"],
+  };
+  
+  const specs = swaggerJsdoc(options);
+  docrouter.use(
+    "/",
+    swaggerUi.serve,
+    swaggerUi.setup(specs)
+  );
+  export default docrouter
```
Comment (GuevaraMarius) [inline documentation/swaggerDoc.js]: Good work Egide, can you use prettier to format your files. Github is requesting a new line at the end of the file.

File: package.json (hunk 1 above)
Comment (GuevaraMarius) [inline package.json]: same here

File: src/controllers/busControllers.js (hunk 2 above)
Comment (GuevaraMarius) [inline src/controllers/busControllers.js]: same here

**Review by GuevaraMarius — APPROVED** [review]

File: src/helpers/socket.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  console.log('User connected');
+  socket.emit('sendToUser', { Bus: `${BusesInfo}` });
+     
+    socket.on('receivedFromUser', function (data) {
+        console.log(data);
+    });
+  socket.on("disconnect", () => {
+      console.log("user disconnected");
+    });
+});
```
Comment (GuevaraMarius) [inline src/helpers/socket.js]: same here

File: src/database/migrations/20201214195024-create-bus.js [hunk 6]
```diff
@@ -16,10 +16,27 @@ module.exports = {
...
+      },
+      status: {
+        type: Sequelize.STRING
+      },
+      commuters: {
+        type: Sequelize.INTEGER
+      },
+      type: {
+        type: Sequelize.STRING
+      },
```
Comment (raymond42) [inline src/database/migrations/20201214195024-create-bus.js]: can we have the same attributes as the ones we have in the model please?

File: src/database/seeders/20201222074608-demo-bus.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+   await queryInterface.bulkInsert('Buses', [
+     {
+       brand:'KBS',
+       plateNo: 'RAB200',
+       driver:'Issa',
```
Comment (raymond42) [inline src/database/seeders/20201222074608-demo-bus.js]: remove the driver. we are no longer assigning the driver when creating the bus

File: src/helpers/socket.js (hunk 5 above)
Comment (raymond42) [inline src/helpers/socket.js]: new line please

**Review by raymond42 — CHANGES_REQUESTED** [review]
attend to the comments and get back to me

**Review by 41hulk — COMMENTED** [review]
Nice Job Egide, i think you can work on the required changes @GuevaraMarius and @raymond42 Requested. LGTM

File: src/controllers/busControllers.js [hunk 8]
```diff
@@ -131,3 +135,27 @@ export const deleteBus = async (req, res) => {
...
   }
 };
+export const busesInfo= async (req,res)=>{
+  try{
+    const response= await Bus.findAll({
+      attributes: ['id','location','status','commuters','type','seats']
+    })
+    successRes(res,200,res.__('All information you need about the bus'),response)
+  }catch(error){
+    console.log(error)
```
Comment (KundwaBruno) [inline src/controllers/busControllers.js]: Great work Egide!, You can remove this console log.

File: src/controllers/busControllers.js [hunk 9]
```diff
@@ -131,3 +135,27 @@ export const deleteBus = async (req, res) => {
...
+  export const onebusInfo= async (req,res)=>{
+    try{
+      const {id}=req.params;
+      const response= await Bus.findOne({ where: { id },attributes: ['id','location',
+      'status','commuters','type','seats'] }
+          
+        );
+      successRes(res,200,res.__('All information you need about the bus'),response)
+    }catch(error){
+      console.log(error)
```
Comment (KundwaBruno) [inline src/controllers/busControllers.js]: Same as here, Remove the console log

File: src/helpers/socket.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+
+const socket = require("socket.io");
```
Comment (KundwaBruno) [inline src/helpers/socket.js]: You can use ES6 syntaxes

File: src/helpers/socket.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+const io=socket(server)
+const BusesInfo= 'I am tracking KBS!'
+io.on('connection', (socket) => {
+  console.log('User connected');
+  socket.emit('sendToUser', { Bus: `${BusesInfo}` });
+     
+    socket.on('receivedFromUser', function (data) {
+        console.log(data);
```
Comment (KundwaBruno) [inline src/helpers/socket.js]: Same as here, remove the console logs

File: src/test/bus.test.js [hunk 12]
```diff
@@ -83,11 +83,34 @@ describe('Bus CRUD tesing', async () => {
       .set('auth', token)
       .send(mockBus);
     const busID = res1.body.Bus.id;
+    console.log(busID)
```
Comment (KundwaBruno) [inline src/test/bus.test.js]: Same as here

**Review by samuel776 — APPROVED** [review]

File: src/database/migrations/20201214195024-create-bus.js [hunk 13]
```diff
@@ -16,10 +16,26 @@ module.exports = {
...
         unique: true
       },
+      
       seats: {
         type: Sequelize.INTEGER,
         defaultValue: 0
       },
+      location: {
+        type: Sequelize.STRING,
+        defaultValue:""
```
Comment (raymond42) [inline src/database/migrations/20201214195024-create-bus.js]: instead of having `defaultValue: ""` we can replace it with `allowNull: true`

**Review by samuel776 — APPROVED** [review]

**Review by GuevaraMarius — APPROVED** [review]

**Review by raymond42 — APPROVED** [review]
LGTM

**Pull request merged**